src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -92,6 +92,7 @@
     <PropertyGroup Condition="'$(_SkipAnalyzers)' == '' and
                               '$(IsImplicitlyTriggeredBuild)' == 'true' and
                               '$(UsingMicrosoftNETSdk)' == 'true' and
+                              '$(OptimizeImplicitlyTriggeredBuild)' != 'false' and
```

Comment:
Do we really need the double ` != 'false'` and ` == 'true'` for `OptimizeImplicitlyTriggeredBuild`? 

I get that it could be a different value altogether, but generally we just check for either false or true, then assume anything else represents the opposite.

Reply:
true means force optimize
false means never optimize
empty means default behavior (rely on TreatWarningsAsErrors)